lib/galaxy/model/migrations/alembic/versions_gxy/e93c5d0b47a9_alter_column_external_user_id.py

```diff
@@ -0,0 +1,43 @@
+"""alter column CustosAuthnzToken.external_user_id to increase length
+
+Revision ID: e93c5d0b47a9
+Revises: e0561d5fc8c7
+Create Date: 2023-10-08 12:11:02.024669
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+from galaxy.model.migrations.util import (
+    alter_column,
+    transaction,
+)
+
+# revision identifiers, used by Alembic.
+revision = "e93c5d0b47a9"
+down_revision = "e0561d5fc8c7"
+branch_labels = None
+depends_on = None
+
+table_name = "custos_authnz_token"
+column_name = "external_user_id"
+
+SHORTER_LENGTH = 64
+LONGER_LENGTH = 128
+
+
+def upgrade():
+    alter_column(table_name, column_name, existing_type=sa.String(SHORTER_LENGTH), type_=sa.String(LONGER_LENGTH))
+
+
+def downgrade():
+    with transaction():
+        # Update any existing rows which currently have a external_user_id value that is too long to be set to NULL.
+        table = sa.sql.table(table_name, sa.Column(column_name, sa.String(LONGER_LENGTH), nullable=True))
+        op.execute(
+            table.update()
+            .where(sa.func.length(table.c.external_user_id) > SHORTER_LENGTH)
+            .values(external_user_id=None)
+        )
```

Comment:
I'm not sure this is the best strategy for downgrading existing rows that may have already a longer value. What will be best to do here?
- Set the value to null (or truncate) and change the `expiration_time` to "expired"?
- Completely delete the row?
- Do nothing when downgrading so we never lose the data?
- Something else?

Reply:
"Do nothing" sounds good as long as no other place expects the short version? Would be nice to not drop their ID if admin has to downgrade.

Reply:
Thank you @martenson, this certainly seems like the most reasonable solution.
I removed the downgrade action and tested it back and forth upgrading and downgrading, and seems to work fine i.e. maintaining the 128 length once you upgrade once.